src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs

```diff
@@ -130,54 +123,105 @@ int IVsEditorFactory.Close()
-                    if (designerLoader is null)
-                    {
-                        goto case "Code";
-                    }
+                    (pbstrEditorCaption, pguidCmdUI, ppunkDocView) = GetWinFormsDesignerData(vsHierarchy, itemid, textBuffer, readOnlyStatus, initializeView: true);
+                    break;
 
-                    try
-                    {
-                        designerLoader.Initialize(_oleServiceProvider, vsHierarchy, (int)itemid, (IVsTextLines)textBuffer);
-                        pbstrEditorCaption = designerLoader.GetEditorCaption((int)readOnlyStatus);
+                case "Code":
 
-                        var designer = designerService.CreateDesigner(_oleServiceProvider, designerLoader);
-                        ppunkDocView = Marshal.GetIUnknownForObject(designer.View);
-                        pguidCmdUI = designer.CommandGuid;
-                    }
-                    catch
+                    (pbstrEditorCaption, pguidCmdUI, ppunkDocView) = GetCodeEditorData(textBuffer, readOnlyStatus, initializeView: true);
+                    break;
+
+                default:
+
+                    return VSConstants.E_INVALIDARG;
+            }
+
+            ppunkDocData = Marshal.GetIUnknownForObject(textBuffer);
+
+            return VSConstants.S_OK;
+        }
+
+        public object GetDocumentData(uint grfCreate, string pszMkDocument, IVsHierarchy pHier, uint itemid)
+        {
+            var editorAdaptersFactoryService = _componentModel.GetService<IVsEditorAdaptersFactoryService>();
+
+            var contentTypeRegistryService = _componentModel.GetService<IContentTypeRegistryService>();
+            var contentType = contentTypeRegistryService.GetContentType(ContentTypeName);
+            IVsTextBuffer? textBuffer = editorAdaptersFactoryService.CreateVsTextBufferAdapter(_oleServiceProvider, contentType);
+
+            if (_encoding)
+            {
+                if (textBuffer is IVsUserData userData)
+                {
+                    // The editor shims require that the boxed value when setting the PromptOnLoad flag is a uint
+                    var hresult = userData.SetData(
+                        VSConstants.VsTextBufferUserDataGuid.VsBufferEncodingPromptOnLoad_guid,
+                        (uint)__PROMPTONLOADFLAGS.codepagePrompt);
+
+                    if (ErrorHandler.Failed(hresult))
                     {
-                        designerLoader.Dispose();
-                        throw;
+                        Marshal.ThrowExceptionForHR(hresult);
                     }
+                }
+            }
 
-                    break;
+            return textBuffer;
+        }
 
-                case "Code":
+        public object GetDocumentView(uint grfCreate, string pszPhysicalView, IVsHierarchy pHier, IntPtr punkDocData, uint itemid)
+        {
+            throw new NotImplementedException();
```

Comment:
Will this be implemented someday, or is it fine for it never to be implemented?

Reply:
There is no reason presently for an implementation as there is no scenario to make this call. So, I didn't want to add code to this method and instead leave it to be implemented when we have callers.

Reply:
So if nothing is calling this, the why is the method in the interface in the first place? Or am I misunderstanding something?

Reply:
It has been added for completion mostly. Most designers are able to return a view and docdata without initializing the editor. There can be scenarios where we need one or the other.